src/infer_request.cc

```diff
@@ -1180,30 +1180,40 @@ InferenceRequest::Normalize()
       const auto& data_type = input.DType();
       const auto& input_dims = input.ShapeWithBatchDim();
       int64_t expected_byte_size = INT_MAX;
+
+      // Skip byte size validation for TensorRT backend because it breaks
+      // shape-size assumption
       bool skip_byte_size_check = false;
-      // Because Triton expects STRING type to be in special format
-      // (prepend 4 bytes to specify string length), so need to add all the
-      // first 4 bytes for each element to find expected byte size
-      if (data_type == inference::DataType::TYPE_STRING) {
-        RETURN_IF_ERROR(
-            ValidateBytesInputs(input_id, input, &expected_byte_size));
-        // FIXME: -1 is used as a signal to skip total byte size validation for
-        // unhandled cases in ValidateBytesInputs.
-        skip_byte_size_check = (expected_byte_size == -1);
-      } else {
-        expected_byte_size = triton::common::GetByteSize(data_type, input_dims);
-      }
+      constexpr char trt_prefix[] = "tensorrt_";
+      const std::string& platform = model_raw_->Config().platform();
+      skip_byte_size_check |= (platform.rfind(trt_prefix) == 0);
+
+      if (!skip_byte_size_check) {
+        // Because Triton expects STRING type to be in special format
+        // (prepend 4 bytes to specify string length), so need to add all the
+        // first 4 bytes for each element to find expected byte size
+        if (data_type == inference::DataType::TYPE_STRING) {
+          RETURN_IF_ERROR(
+              ValidateBytesInputs(input_id, input, &expected_byte_size));
+          // FIXME: -1 is used as a signal to skip total byte size validation
+          // for unhandled cases in ValidateBytesInputs.
+          skip_byte_size_check |= (expected_byte_size == -1);
+        } else {
+          expected_byte_size =
+              triton::common::GetByteSize(data_type, input_dims);
+        }
 
-      bool byte_size_valid =
-          (byte_size > INT_MAX) ||
-          (static_cast<int64_t>(byte_size) != expected_byte_size);
-      if (!skip_byte_size_check && byte_size_valid) {
-        return Status(
-            Status::Code::INVALID_ARG,
-            LogRequest() + "input byte size mismatch for input '" + input_id +
-                "' for model '" + ModelName() + "'. Expected " +
-                std::to_string(expected_byte_size) + ", got " +
-                std::to_string(byte_size));
+        bool byte_size_valid =
+            (byte_size > INT_MAX) ||
+            (static_cast<int64_t>(byte_size) != expected_byte_size);
+        if (!skip_byte_size_check && byte_size_valid) {
```

Comment:
This was my mistake, but let's rename `byte_size_valid` to `byte_size_invalid` or similar.